### ISSUE TITLE
Missing info on data frame structure/contents for iPSC analysis

### ISSUE THREAD

**Issue by jdee3** [anna-saukkonen/PAC#9]
Hello,

Thanks for this great resource!

I've been trying to reproduce your iPSC analysis (hipsci_analysis.py), but I'm having trouble running the code because it isn't clear what the columns are for the input data frames (i.e. ipsc_df). Specifically:

-  You refer to 'MapCov' and 'TotRead' columns, but TotRead isn't a listed column in any of the PAC output files...so how has this obtained? Are MapCov and TotRead the same?
- For ASE selection, the function ASE_selection filters based on a 'p' column in the data frame. Since the binomial test is not run within PAC, was the binomial test run for each row of the PAC txt file (results_2genomes_ID.RSEM.STAR.SOFT.NOTRIM_baq.txt), creating this 'p' column?
- Is the Chr_Pos column created after the PAC run?

Thanks again!

**Comment by AJHodgkinson** [comment]
Hi jdee3,

Thanks for your interest in the pipeline.  To answer your questions:

1. PAC will produce 2 types of output file, a haplotype (gene) level one, and a site-level one.  In the gene level output there is a column called 'totalCount' which is the total coverage. In the site level one, there is 'Mapcov' which is the total coverage. The main page has instructions on each column in the two types of output file.
2. Yes, we ran a binomial test for every row in the output file for this analysis.
3. We don't create 'Chr_pos' column from the output of PAC, we just generated this in later analyses.

I hope that helps, but please get in touch if something is still unclear.

**Comment by jdee3** [comment]
Thanks for the prompt reply AJ.

Just to clarify- the code defines the following function:

    def columns_to_numeric_and_filter_coverage(df):
        """
        Changes relevant columns to numeric types.
        """
        df['Chr'] = df['Chr'].astype(int)
        df['Pos'] = df['Pos'].astype(int)
        df['MapCov'] = df['MapCov'].astype(float)
        df['TotRead'] = df['TotRead'].astype(int)
        df = df.loc[df.TotRead >= 10]

How does this data frame, df, contain both MapCov (from the site-level output) and total counts (from the gene-level output) when the rows in these text files don't match? How was this df created, is basically what I'm trying to disentangle. Seems to be a fusion of info from the output files? How did you extract total counts for each site, given that you only have total counts per gene? Of course, going the other way is easy, as you can sum the map coverage for all sites within a gene, but how do you divide the gene total counts across all the sites?

Thanks a lot for your help!

**Comment by AJHodgkinson** [comment]
Hi jdee,

Sorry for the delay - this script was written by Anna and she has now left the lab.  I believe that the script you are referring to is from Anna's thesis (if it is in the thesis directory), rather than the manuscript. Which analysis are you trying to reproduce?  Is it the GxE analysis of cells treated to different environmental conditions (Findley data)?

**Comment by jdee3** [comment]
Yes, the file 'hipsci_analysis.py' is in the thesis_scripts folder. Wanted to generate the same kinds of plots with the data, but the data frame structure isn't quite clear, so I was trying to figure that out.

**Comment by anna-saukkonen** [comment]
Hello,

Apologies for late reply, I haven't kept in touch as well after leaving the lab. I changed the column names and forgot to clarify on the GitHub. All the information needed to reproduce the df in the hipsci_analysis.py is in the PAC output. MapCov refers to Ref allele coverage, TotRead to total number of reads covering the site, Chr refers to chromosome and Pos to position along the chromosome. They do not refer to different output files, but within file (eg. site level will contain all these columns just renamed and some columns that I didn't need were deleted) I renamed for old convention we used for unpublished data as previous PAC version had different output. Apologies for the confusion.
